wp-admin/js/widgets/media-image-widget.js

```diff
@@ -25,17 +25,41 @@
 	ImageWidgetControl = component.MediaWidgetControl.extend({
 
 		/**
+		 * Get the appropriate image url for the preview.
+		 *
+		 * @returns {string} Image url
+		 */
+		getImagePreviewUrl: function getImagePreviewUrl() {
+			var control = this, attachmentId, size, attachmentSizes;
+			attachmentId = control.model.get( 'attachment_id' );
+			size = control.model.get( 'size' );
+
+			// Attempt to return the specific attachment image size requested.
+			attachmentSizes = control.selectedAttachment.get( 'sizes' );
+			if ( attachmentId && 'custom' !== size && attachmentSizes && attachmentSizes[ size ] ) {
+				return attachmentSizes[ size ].url;
+			}
```

Comment:
Yeah I thought the same thing but in my testing I was running into scenarios when the image url was not respecting the size chosen.  Let me take it for another spin and see if I can reproduce/track down the root cause if that is the case.